src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -293,4 +318,45 @@ class BlobCacheServiceImpl implements BlobCacheService {
             }
         }
     }
+
+   /**
+    * get the size of the blob stored in the cache
+    *
+    * @return {@link Long} the size of the blob stored in the cache
+    */
+   protected Long getBlobSize(String key) {
+        try {
+            final headObjectRequest =
+                    HeadObjectRequest.builder()
+                            .bucket(BucketTokenizer.from(blobConfig.storageBucket).bucket)
+                            .key(key)
```

Comment:
In principle, this should not work if uri is `s3://foo/bar` the bucket is `foo` and the key `bar` (without slashes). Also better to renamed the method arg to `path` or something else

Reply:
I have added bucketwithpath method to get the URI without protocol
Sorry, I don't understand which method you is arg

Reply:
Stating the documentation `.bucket()` should take only the bucket name (not the path) and `.key()` the path without the bucket name

Reply:
done